packages/react/src/themes/base/components/Icon/iconStyles.ts

```diff
@@ -0,0 +1,136 @@
+  largest: 40,
+}
+
+const getDefaultFontIcon = (iconName: string) => {
+  return callable(fontAwesomeIcons(iconName).icon)()
+}
+
+const getFontStyles = (
+  size: number,
+  iconName: string,
+  themeIcon?: ResultOf<FontIconSpec>,
+): ICSSInJSStyle => {
+  const { fontFamily, content } = themeIcon || getDefaultFontIcon(iconName)
+  const sizeInRems = pxToRem(size)
+
+  return {
+    fontFamily,
+    fontSize: sizeInRems,
+    lineHeight: 1,
+
+    display: 'inline-flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+
+    width: sizeInRems,
+    height: sizeInRems,
+
+    '::before': {
+      content,
+    },
+  }
+}
+
+const getXSpacingStyles = (xSpacing: IconXSpacing, horizontalSpace: string): ICSSInJSStyle => {
+  switch (xSpacing) {
+    case 'none':
+      return { marginLeft: 0, marginRight: 0 }
+    case 'before':
+      return { marginLeft: horizontalSpace, marginRight: 0 }
+    case 'after':
+      return { marginLeft: 0, marginRight: horizontalSpace }
+    case 'both':
+      return { marginLeft: horizontalSpace, marginRight: horizontalSpace }
+  }
+}
+
+const getBorderedStyles = (boxShadowColor: string): ICSSInJSStyle => {
+  return {
+    ...getPaddedStyle(),
+
+    boxShadow: `0 0 0 .05rem ${boxShadowColor} inset`,
+  }
+}
+
+const getPaddedStyle = (): ICSSInJSStyle => ({
+  padding: pxToRem(4),
+})
+
+const getIconSize = (size: SizeValue, sizeModifier: IconSizeModifier): number => {
+  if (!sizeModifier) {
+    return sizes[size]
+  }
+
+  const modifiedSizes = {
+    large: {
+      x: 24,
+      xx: 28,
+    },
+  }
+
+  return modifiedSizes[size] && modifiedSizes[size][sizeModifier]
+}
+
+const getIconColor = (colorProp: string, variables: IconVariables) =>
+  _.get(variables.colors, colorProp, variables.color || 'currentColor')
+
+const iconStyles: ComponentSlotStylesInput<IconProps, IconVariables> = {
+  root: ({
+    props: { disabled, name, size, bordered, circular, color, xSpacing, rotate },
+    variables: v,
+    theme,
+  }): ICSSInJSStyle => {
+    const iconSpec = theme.icons[name]
+    const rtl = theme.rtl
+    const isFontBased = name && (!iconSpec || !iconSpec.isSvg)
+
+    return {
+      backgroundColor: v.backgroundColor,
+      display: 'inline-block',
+      speak: 'none',
+      verticalAlign: 'middle',
+
+      ...(!isFontBased && { boxSizing: 'border-box' }),
+
+      ...(isFontBased && getFontStyles(getIconSize(size, v.sizeModifier), name)),
+
+      ...(isFontBased && {
+        color: getIconColor(color, v),
+        fontWeight: 900, // required for the fontAwesome to render
+
+        ...(disabled && {
+          color: v.disabledColor,
+        }),
+      }),
+
+      ...getXSpacingStyles(xSpacing, v.horizontalSpace),
+
+      ...(circular && { ...getPaddedStyle(), borderRadius: '50%' }),
+
+      ...((bordered || v.borderColor) &&
+        getBorderedStyles(v.borderColor || getIconColor(color, v))),
+
+      ...(rtl && {
+        transform: `scaleX(-1) rotate(${-1 * rotate}deg)`,
```

Comment:
just to be sure - seems like `rotate` prop's value won't have any effect in case if `rtl` mode is disabled, will it?

Reply:
yep, will add it